recipes/cppcheck/all/conanfile.py

```diff
@@ -14,37 +13,34 @@ class CppcheckConan(ConanFile):
     topics = ("code quality", "static analyzer", "linter")
     description = "Cppcheck is an analysis tool for C/C++ code."
     license = "GPL-3.0-or-later"
+    package_type = "application"
     settings = "os", "arch", "compiler", "build_type"
-    options = {"with_z3": [True, False], "have_rules": [True, False]}
-    default_options = {"with_z3": True, "have_rules": True}
+    options = {"have_rules": [True, False], "with_z3": [True, False, "deprecated"]}
```

Comment:
?? This option is not deprecated in the sense of conan recipe, it was already removed for ccpcheck version >= 2.8.0

Reply:
Yes, that's right. I only implemented what I was told.

Reply:
Sadly when 2.8 was added it was never marked as deprecated.
This is the first one to drop less then that so in a few weeks feel free to delete the option entirely :)